packages/rum-core/src/domain/rumEventsCollection/action/trackActions.ts

```diff
@@ -250,3 +270,52 @@ function newPotentialAction(
     },
   }
 }
+
+function addClickToClickChain(state: TrackActionsState, singleClickPotentialAction: PotentialAction): ClickReference {
+  const click: ActionClick = {
+    event: singleClickPotentialAction.base.event,
+    timeStamp: singleClickPotentialAction.base.startClocks.timeStamp,
+    singleClickPotentialAction,
+  }
+  let clickReference = state.currentClickChain && state.currentClickChain.tryAppend(click)
+  // If we failed to add the click to the current click chain, create a new click chain
+  if (!clickReference) {
+    const rageClickPotentialAction = newPotentialAction(state, singleClickPotentialAction.base)
+    const newClickChain = createClickChain<ActionClick>((clicks) => {
+      flushClickChain(clicks, rageClickPotentialAction)
+      stopSubscription.unsubscribe()
+    })
+    const stopSubscription = state.stopObservable.subscribe(newClickChain.stop)
+    // adding a click to a newly created click chain should always succeed
+    clickReference = newClickChain.tryAppend(click)!
+    state.currentClickChain = newClickChain
+  }
+  return clickReference
+}
+
+function flushClickChain(clicks: ActionClick[], rageClickPotentialAction: PotentialAction) {
+  if (isRage(clicks)) {
+    // Merge any click frustration to the rage click action. In practice, it only concerns
+    // 'dead', because any potential 'error' frustration will already be there (as potential action
+    // collect it themselves), and 'rage' won't be on single click potential actions.
+    clicks.forEach((click) => {
+      click.singleClickPotentialAction.frustrations.forEach((frustration) =>
+        rageClickPotentialAction.frustrations.add(frustration)
+      )
+    })
+    // Send the rage click action
+    rageClickPotentialAction.frustrations.add(FrustrationType.RAGE)
+    rageClickPotentialAction.complete(timeStampNow())
+    rageClickPotentialAction.notifyIfComplete()
+  } else {
+    rageClickPotentialAction.discard()
+    // Send an action for each individual click
+    clicks.forEach((click) => click.singleClickPotentialAction.notifyIfComplete())
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏`notifyIfComplete` seems to be referenced as `send` in your comments, there is maybe something that could be improved around that.
Proposition: call `complete` only for the actions that we will send as `AUTO_ACTION_COMPLETED`.
We could have a `action.stop(endTime)` and `action.complete(endTime?)` (which stop the action if it is not already stopped) allowing us to:

- stop potential actions and complete them later if they are not part of a rage click
- complete directly rage click actions instead of having to call two APIs